src/fsharp/Interactive.DependencyManager/DependencyManager.fs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Interactive.DependencyManager
+
+open System
+open System.IO
+open System.Reflection
+open FSharp.Reflection
+open Internal.Utilities.FSharpEnvironment
+
+module ReflectionHelper =
+    let dependencyManagerPattern = "*DependencyManager*.dll"
+    let dependencyManagerAttributeName= "DependencyManagerAttribute"
+    let resolveDependenciesMethodName = "ResolveDependencies"
+    let namePropertyName = "Name"
+    let keyPropertyName = "Key"
+    let seqEmpty = Seq.empty<string>
+
+    let assemblyHasAttribute (theAssembly: Assembly) attributeName =
+        try
+            CustomAttributeExtensions.GetCustomAttributes(theAssembly)
+            |> Seq.exists (fun a -> a.GetType().Name = attributeName)
+        with | _ -> false
```

Comment:
This is the first thing going on with reflection and it may fail due to error such as "couldn't load FSharp.Core" due to missing binding redirect.

Should the error be reported, like "error while checking assembly assemblyname: exception message"?